src/test/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsTest.java

```diff
@@ -0,0 +1,138 @@
+
+    @Mock
+    private CertificateExpiryMonitor mockCertExpiryMonitor;
+
+    @Mock
+    private CISShadowMonitor mockShadowMonitor;
+
+    @Mock
+    private SecurityService securityServiceMock;
+
+    @Mock
+    private GreengrassServiceClientFactory clientFactoryMock;
+
+    @TempDir
+    Path tmpPath;
+
+    private CertificateManager certificateManager;
+    private CertificateStore certificateStore;
+    private ClientDeviceAuthMetrics metrics;
+    CertificateRotationHandler certRotationMonitor;
+
+    @BeforeEach
+    void beforeEach() {
+        DomainEvents domainEvents = new DomainEvents();
+        metrics = new ClientDeviceAuthMetrics();
+        certificateStore = spy(new CertificateStore(tmpPath, domainEvents, securityServiceMock));
+        certRotationMonitor = new CertificateRotationHandler(mockConnectivityInformation, domainEvents, metrics);
+
+        certificateManager = new CertificateManager(certificateStore, mockConnectivityInformation,
+                mockCertExpiryMonitor, mockShadowMonitor, Clock.systemUTC(), clientFactoryMock, securityServiceMock,
+                certRotationMonitor, metrics);
+
+        CertificatesConfig certificatesConfig =
+                new CertificatesConfig(Topics.of(new Context(), CONFIGURATION_CONFIG_KEY, null));
+        certificateManager.updateCertificatesConfiguration(certificatesConfig);
+
+        emitterMock = new MetricsEmitter(metrics);
+    }
+
+    @Test
+    void GIVEN_defaultCertManager_WHEN_subscribeToCertificateUpdates_THEN_subscribeSuccessMetricCorrectlyEmitted()
+            throws KeyStoreException, CertificateGenerationException {
+        certificateManager.generateCA("", CertificateStore.CAType.RSA_2048);
+
+        Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> con = TestUtils.asyncAssertOnConsumer((a) -> {
+        }, 3);
+
+        GetCertificateRequestOptions requestOptions = new GetCertificateRequestOptions();
+        requestOptions.setCertificateType(GetCertificateRequestOptions.CertificateType.SERVER);
+        GetCertificateRequest certificateRequest =
+                new GetCertificateRequest("testService", requestOptions, con.getRight());
+
+        // Subscribe multiple times to ensure the Cert.SubscribeSuccess metric increments correctly
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        assertEquals(3L, metrics.getSubscribeSuccess());
+
+        // Checking that the emitter collects the metrics as expected
+        Metric metric = Metric.builder()
+                .namespace("ClientDeviceAuth")
+                .name("Cert.SubscribeSuccess")
+                .unit(TelemetryUnit.Count)
+                .aggregation(TelemetryAggregation.Sum)
+                .value(3L)
+                .timestamp(Instant.now().toEpochMilli())
```

Comment:
since we're comparing timestamps below, there's a race between this and https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/afc52b6e0be2121fdb2620f3d412b7b3a6c64c7b/src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java#L30

you could pass in a clock to `ClientDevicesAuthMetrics` and have instants in that class and in this test method use the same clock